main_test.go

```diff
@@ -81,8 +81,14 @@ var (
 )
 
 // lookup all the request metrics in the database paging as necessary
-// search for any request metrics during the time window for particular request methods
-func findMetricsInWindowForMethods(db database.PostgresClient, startTime time.Time, endTime time.Time, testedmethods []string) []database.ProxiedRequestMetric {
+// search for any request metrics between starTime and time.Now() for particular request methods
+func findMetricsInWindowForMethods(db database.PostgresClient, startTime time.Time, testedmethods []string) []database.ProxiedRequestMetric {
```

Comment:
every occurrence of this helper func used `time.Now()` as the `endTime` so i made that an assumption of the func.
this allows us to secretly sleep for 10 microseconds in order to allow metrics to actually be created.

doing so fixes frequent but random test failures that look like `expected 4 metrics but found 2` or `[] does not contain yada yada`